packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -445,14 +453,22 @@ function navigateReducer_PPR(
             // tree. Or in the meantime we could factor it out into a
             // separate function.
             const cache: CacheNode = createEmptyCacheNode()
+            const hasReusablePrefetch =
+              prefetchValues.kind === 'auto' &&
+              prefetchEntryCacheStatus === PrefetchCacheEntryStatus.reusable
             let applied = applyFlightData(
               currentCache,
               cache,
               flightDataPath,
-              prefetchValues.kind === 'auto' &&
-                prefetchEntryCacheStatus === PrefetchCacheEntryStatus.reusable
+              hasReusablePrefetch
             )
 
+            if (!hasReusablePrefetch) {
+              // Once the prefetch entry no longer reusable, `applyFlightData` will signal to layout router that it needs to lazy fetch the data
+              // We update the `lastUsedTime` so that we renew the 30s cache for this entry
+              prefetchValues.lastUsedTime = Date.now()
```

Comment:
Couldn't this be moved to `getOrCreatePrefetchCacheEntry`?

Reply:
I think the sequencing here matters -- we only want to refresh the entry once `applyFlightData` has operated on the cache since there are conditions in navigate reducer where we bail out of applying the data. I think if we move this into `getOrCreatePrefetchCacheEntry` we'll be inadvertently renewing the cache every read rather than only when data is refreshed from the server.

There might be a way to add some similar checks in `getOrCreatePrefetchCacheEntry` but I think we'd end up some of the existing logic in navigate reducer. Since the reducer already has some logic to update `lastUsedTime` on first read, I figured this was a reasonable spot to do something similar.

Reply:
> there are conditions in navigate reducer where we bail out of applying the data

I'm actually not sure this makes sense, though, because we're using the prefetch cache data itself to determine whether to bail out. So it creates this contradiction where we could be using a stale prefetch cache entry to determine whether or not the data can be applied. So something's not adding up. If it's stale, then we shouldn't be allowed to use any of the information in the payload. Which implies that this staleness check should come before the `data.then`.